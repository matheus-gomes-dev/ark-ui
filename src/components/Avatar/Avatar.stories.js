import React from 'react';

import Avatar from './';


export default {
  title: 'Example/Avatar',
  component: Avatar
};

const Template = (args) => <Avatar {...args} />;

export const User = Template.bind({});
User.args = {
  type: 'user',
};

export const Wallet = Template.bind({});
Wallet.args = {
  type: 'wallet',
};