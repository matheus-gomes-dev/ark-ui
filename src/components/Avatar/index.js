import React from 'react';
import PropTypes from 'prop-types';

import AvatarIcon from 'assets/avatar.png';
import WalletIcon from 'assets/wallet.png';

const Avatar = ({ name = 'Matheus Gomes', amount = '0.00', type = 'user' }) => (
  <div className="h-full flex justify-center items-center">
    <div
      className="bg-cover bg-center h-24 w-24 rounded-full"
      style={{ backgroundImage: `url(${type === 'user' ? AvatarIcon : WalletIcon})`}}
    />
    <div className="flex flex-col justify-center items-left pl-4">
      <span className={type === 'user' ? "font-bold text-xl text-black" : "text-sm text-gray-600"}>{name}</span>
      <span className={type === 'user' ? "text-sm text-gray-600" : "font-bold text-xl text-black"}>
        {type === 'user' ? `Total Balance: ${amount}` : amount}
      </span>
    </div>
  </div>
);


Avatar.propTypes = {
  name: PropTypes.string,
  amount: PropTypes.string,
  type: PropTypes.oneOf(['user', 'avatar'])
};

export default Avatar;
