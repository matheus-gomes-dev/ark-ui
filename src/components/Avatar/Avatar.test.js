import React from 'react';
import renderer from 'react-test-renderer';

import Avatar from './';

describe('Avatar', () => {
  
  it('should render default component correctly', () => {
    const tree = renderer.create(<Avatar />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render component correctly when type equals wallet', () => {
    const tree = renderer.create(<Avatar />).toJSON();
    expect(tree).toMatchSnapshot();
  });

})