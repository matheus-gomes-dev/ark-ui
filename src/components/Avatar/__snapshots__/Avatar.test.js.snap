// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar should render component correctly when type equals wallet 1`] = `
<div
  className="h-full flex justify-center items-center"
>
  <div
    className="bg-cover bg-center h-24 w-24 rounded-full"
    style={
      Object {
        "backgroundImage": "url(avatar.png)",
      }
    }
  />
  <div
    className="flex flex-col justify-center items-left pl-4"
  >
    <span
      className="font-bold text-xl text-black"
    >
      Matheus Gomes
    </span>
    <span
      className="text-sm text-gray-600"
    >
      Total Balance: 0.00
    </span>
  </div>
</div>
`;

exports[`Avatar should render default component correctly 1`] = `
<div
  className="h-full flex justify-center items-center"
>
  <div
    className="bg-cover bg-center h-24 w-24 rounded-full"
    style={
      Object {
        "backgroundImage": "url(avatar.png)",
      }
    }
  />
  <div
    className="flex flex-col justify-center items-left pl-4"
  >
    <span
      className="font-bold text-xl text-black"
    >
      Matheus Gomes
    </span>
    <span
      className="text-sm text-gray-600"
    >
      Total Balance: 0.00
    </span>
  </div>
</div>
`;
